src/main/java/pingis/controllers/TaskController.java

```diff
@@ -71,19 +73,19 @@ public String task(Model model,
     model.addAttribute("challenge", currentChallenge);
     model.addAttribute("task", taskInstance.getTask());
     model.addAttribute("taskInstanceId", taskInstanceId);
+
     Map<String, EditorTabData> editorContents = editorService.generateEditorContents(taskInstance);
-    model.addAttribute("submissionCodeStub", editorContents.get("editor1").code);
-    model.addAttribute("staticCode", editorContents.get("editor2").code);
-    String implFileName = JavaClassGenerator.generateImplClassFilename(currentChallenge);
-    String testFileName = JavaClassGenerator.generateTestClassFilename(currentChallenge);
-
-    if (taskInstance.getTask().getType() == TaskType.TEST) {
-      model.addAttribute("submissionTabFileName", testFileName);
-      model.addAttribute("staticTabFileName", implFileName);
-    } else {
-      model.addAttribute("submissionTabFileName", implFileName);
-      model.addAttribute("staticTabFileName", testFileName);
-    }
+    EditorTabData testTab = editorContents.get("editor1");
+    EditorTabData implTab = editorContents.get("editor2");
```

Comment:
Tab names should probably be tab1/tab2, leftTab/rightTab, submissionTab/staticTab or something similar. The editor1 and editor2 names could be changed accordingly as well.